datafusion/core/src/catalog/mod.rs

```diff
@@ -295,6 +296,118 @@ impl CatalogProvider for MemoryCatalogProvider {
     }
 }
 
+/// Resolve all table references in the SQL statement.
+///
+/// ## Example
+///
+/// ```
+/// use datafusion_sql::parser::DFParser;
+/// use datafusion::catalog::resolve_table_references;
+///
+/// let query = "SELECT a FROM foo where x IN (SELECT y FROM bar)";
+/// let statement = DFParser::parse_sql(query).unwrap().pop_back().unwrap();
+/// let table_refs = resolve_table_references(&statement, true).unwrap();
+/// assert_eq!(table_refs.len(), 2);
+/// assert_eq!(table_refs[0].to_string(), "foo");
+/// assert_eq!(table_refs[1].to_string(), "bar");
+/// ```
```

Comment:
```suggestion
/// ```
///
/// ## Example with CTEs
///
/// ```
/// # use datafusion_sql::parser::DFParser;
/// # use datafusion::catalog::resolve_table_references;
/// let query = "with my_cte as (values (1), (2)) SELECT * from my_cte;";
/// let statement = DFParser::parse_sql(query).unwrap().pop_back().unwrap();
/// let table_refs = resolve_table_references(&statement, true).unwrap();
/// assert_eq!(table_refs.len(), 1);
/// assert_eq!(table_refs[0].to_string(), "my_cte");
/// ```
```

Reply:
Thanks this is very handy, I've added it.